tests/BuildScriptGenerator.Tests/Python/PythonPlatformTests.cs

```diff
@@ -14,11 +14,99 @@ namespace Microsoft.Oryx.BuildScriptGenerator.Tests.Python
+            var environment = new TestEnvironment();
+            var installerScriptSnippet = "##INSTALLER_SCRIPT##";
+            var versionProvider = new TestPythonVersionProvider(new[] { "3.7.5", "3.8.0" }, defaultVersion: "3.7.5");
+            var platformInstaller = new TestPythonPlatformInstaller(
+                isVersionAlreadyInstalled: false,
+                installerScript: installerScriptSnippet,
+                Options.Create(options),
+                environment);
+            var platform = CreatePlatform(environment, versionProvider, platformInstaller, options);
+            var repo = new MemorySourceRepo();
+            repo.AddFile("", PythonConstants.RequirementsFileName);
+            repo.AddFile("print(1)", "bla.py");
+            var context = new BuildScriptGeneratorContext { SourceRepo = repo, PythonVersion = "3.7.5" };
+
+            // Act
+            var snippet = platform.GenerateBashBuildScriptSnippet(context);
+
+            // Assert
+            Assert.NotNull(snippet);
+            Assert.NotNull(snippet.InstallationScriptSnippet);
+            Assert.Equal(installerScriptSnippet, snippet.InstallationScriptSnippet);
+            Assert.Contains(ManifestFilePropertyKeys.PythonVersion, snippet.BuildProperties.Keys);
+            Assert.Equal("3.7.5", snippet.BuildProperties[ManifestFilePropertyKeys.PythonVersion]);
+        }
+
+        [Fact]
+        public void GeneratedSnippet_DoesNotHaveInstallScript_IfVersionIsAlreadyPresentOnDisk()
+        {
+            // Arrange
+            var options = new BuildScriptGeneratorOptions() { EnableDynamicInstall = true };
+            var environment = new TestEnvironment();
+            var installerScriptSnippet = "##INSTALLER_SCRIPT##";
+            var versionProvider = new TestPythonVersionProvider(new[] { "3.7.5", "3.8.0" }, defaultVersion: "3.7.5");
+            var platformInstaller = new TestPythonPlatformInstaller(
+                isVersionAlreadyInstalled: true,
+                installerScript: installerScriptSnippet,
+                Options.Create(options),
+                environment);
+            var platform = CreatePlatform(environment, versionProvider, platformInstaller, options);
+            var repo = new MemorySourceRepo();
+            repo.AddFile("", PythonConstants.RequirementsFileName);
+            repo.AddFile("print(1)", "bla.py");
+            var context = new BuildScriptGeneratorContext { SourceRepo = repo, PythonVersion = "3.7.5" };
+
+            // Act
+            var snippet = platform.GenerateBashBuildScriptSnippet(context);
+
+            // Assert
+            Assert.NotNull(snippet);
+            Assert.Null(snippet.InstallationScriptSnippet);
+            Assert.Contains(ManifestFilePropertyKeys.PythonVersion, snippet.BuildProperties.Keys);
+            Assert.Equal("3.7.5", snippet.BuildProperties[ManifestFilePropertyKeys.PythonVersion]);
+        }
+
```

Comment:
is it possible to add a test that downloading platform binaries works behind a proxy?